dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/util/ContentletUtilTest.java

```diff
@@ -64,251 +70,138 @@ public static void prepare () throws Exception {
+                    .values(contentBeltsCategory.getInode()).next());
+            contentType = new ContentTypeDataGen().host(host).fields(fields).nextPersisted();
 
             //Creating content.
-            contentlet = new Contentlet();
-            contentlet.setStructureInode( contentType.inode() );
-            contentlet.setLanguageId( language.getId() );
-            contentlet.setStringProperty( textField.variable(), "Test Contentlet" );
-
-            List<Category> contentletCategories = Arrays.asList( popularCategory, homeCategory );
-
-            contentletAPI.validateContentlet( contentlet, contentletCategories  );
-            contentlet.setIndexPolicy(IndexPolicy.FORCE);
-            contentlet = contentletAPI.checkin( contentlet, null, contentletCategories, user, false );
-
-            assertTrue( UtilMethods.isSet( contentlet.getInode() ) );
-
-            final Map<String, Object> contentPrintableMap = ContentletUtil.getContentPrintableMap( user, contentlet );
-            assertTrue( contentPrintableMap.containsKey( CATEGORY_NAME_CONTENT ) );
-            assertTrue( contentPrintableMap.containsKey( CATEGORY_NAME_CONTENT_BELTS ) );
-            assertEquals( popularCategory.getCategoryName(), contentPrintableMap.get( CATEGORY_NAME_CONTENT ) );
-            assertEquals( homeCategory.getCategoryName(), contentPrintableMap.get( CATEGORY_NAME_CONTENT_BELTS ) );
-
-        } catch ( Exception e ){
+            contentlet = new ContentletDataGen(contentType.id())
+                    .languageId(language.getId())
+                    .setProperty("title", "Test Contentlet")
+                    .addCategory(popularCategory)
+                    .addCategory(homeCategory).addCategory(flightsCategory)
+                    .nextPersisted();
 
-            fail(e.getMessage());
-
-        } finally {
-
-            //Delete Contentlet.
-            if ( contentlet != null ){
-
-                contentletAPI.destroy(contentlet, user, false);
+            final Map<String, Object> contentPrintableMap = ContentletUtil
+                    .getContentPrintableMap( user, contentlet,
+                            categoryInfoType.equals(ALL_CATEGORIES_INFO));
+
+            assertTrue(contentPrintableMap.containsKey(CATEGORY_NAME_CONTENT));
+            assertTrue(contentPrintableMap.containsKey(CATEGORY_NAME_CONTENT_BELTS));
+            if (categoryInfoType.equals(ALL_CATEGORIES_INFO)) {
+                verifyCategoriesForField(contentPrintableMap, CATEGORY_NAME_CONTENT,
+                        popularCategory);
+                verifyCategoriesForField(contentPrintableMap, CATEGORY_NAME_CONTENT_BELTS,
+                        homeCategory, flightsCategory);
+            } else {
+                assertEquals( popularCategory.getCategoryName(), contentPrintableMap.get( CATEGORY_NAME_CONTENT ) );
+                String [] resultCategories = contentPrintableMap.get(CATEGORY_NAME_CONTENT_BELTS)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'resultCategories' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4885353)